internal/plugin/github.go

```diff
@@ -4,56 +4,56 @@ import (
 	"io"
 	"net/http"
 	"net/url"
-	"strings"
 
-	"github.com/pkg/errors"
+	"github.com/samber/lo"
 	"go.jetpack.io/devbox/internal/boxcli/usererr"
 	"go.jetpack.io/devbox/internal/cachehash"
 )
 
 type githubPlugin struct {
-	raw      string
-	org      string
-	repo     string
-	revision string
-	dir      string
+	RefLike
 }
 
-// newGithubPlugin returns a plugin that is hosted on github.
-// url is of the form org/repo?dir=<dir>
-// The (optional) dir must have a plugin.json"
-func newGithubPlugin(rawURL string) (*githubPlugin, error) {
-	pluginURL, err := url.Parse(rawURL)
+func newGithubPlugin(ref RefLike) *githubPlugin {
+	if ref.Ref.Ref == "" && ref.Rev == "" {
+		ref.Ref.Ref = "master"
+	}
+	return &githubPlugin{RefLike: ref}
+}
+
+func (p *githubPlugin) Fetch() ([]byte, error) {
+	// Github redirects "master" to "main" in new repos. They don't do the reverse
+	// so setting master here is better.
+	contentURL, err := url.JoinPath(
+		"https://raw.githubusercontent.com/",
+		p.Owner,
+		p.Repo,
+		lo.Ternary(p.Rev == "", "master", p.Rev),
+		p.withFilename(p.Dir),
```

Comment:
not sure I follow: why are we passing in `p.Dir` (a directory) as an argument for `withFilename`?

Reply:
different plugin types have different fields represent the path. Local plugins use `Path` while github plugins use `Dir`

Maybe a cleaner solution could be to implement a `configPath()` function for each type. (though they would still both have some duplicate code)